### PR TITLE
[3.9] [BTS-1148] Fix DBServer Race in running Query abort on DBServer

### PR DESCRIPTION
**Pull request by mpoeter**
### Scope & Purpose

Fix a race when aborting/finishing a currently active query on a DB-Server. This race could cause the query to remain in the server's query registry longer than indented, potentially holding some locks. Such queries were garbage collected eventually, but this could take a while, depending on the specified TTL (10min per default). This has now been fixed so that aborted/finished queries are cleaned up in a timely manner.

- [x] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] **integration tests**
- [x] Backports
  - [x] Backport for devel: #18391
  - [x] Backport for 3.10: #18395

#### Related Information

- [x] Jira ticket: https://arangodb.atlassian.net/browse/BTS-1148

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,14 @@
 v3.9.10 (XXXX-XX-XX)
 --------------------
 
+* BTS-1148: Fix a race when aborting/finishing a currently active query
+  on a DB-Server. This race could cause the query to remain in the server's
+  query registry longer than indented, potentially holding some locks. Such
```
Comment (jsteemann) [inline CHANGELOG:6]:
```suggestion
  query registry longer than intended, potentially holding some locks. Such
```

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
set the "do not merge" flag temporarily, because we have seen some of the new tests fail sporadically in devel.

**Review by mchacki — APPROVED** [review]
LGTM

**Pull request merged**